src/main/java/com/yellow/ordermanageryellow/service/ChargingService.java

```diff
@@ -0,0 +1,55 @@
+package com.yellow.ordermanageryellow.service;
+
+import com.yellow.ordermanageryellow.Dao.OrdersRepository;
+import com.yellow.ordermanageryellow.Dao.ProductRepository;
+import com.yellow.ordermanageryellow.Dto.OrderDTO;
+import com.yellow.ordermanageryellow.Dto.OrderMapper;
+import com.yellow.ordermanageryellow.model.Order_Items;
+import com.yellow.ordermanageryellow.model.Orders;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.stereotype.Service;
+
+@Service
+public class ChargingService {
+    @Autowired
+    private OrdersRepository ordersRepository;
+    @Autowired
+    private ProductRepository productRepository;
+    @Autowired
+    private RabbitMQProducer rabbitMQProducer;
+
+    public void chargingStep(Orders order) {
+        Orders orderFromMongo=ordersRepository.findById(order.getId()).orElse(null);
+        try{
+            orderFromMongo.setOrderStatusId(Orders.status.charging);
+            for (Order_Items item:orderFromMongo.getOrderItems())
+            {
+                if(item.getProductId().getInventory()<item.getQuantity()){
+                    orderFromMongo.setOrderStatusId(Orders.status.cancelled);
+                    ordersRepository.save(orderFromMongo);
+                break;}
+                else{
+                    item.getProductId().setInventory((int)(item.getProductId().getInventory()-item.getQuantity()));
+                    OrderDTO orderDTO = OrderMapper.INSTANCE.orderToOrderDTO(orderFromMongo);
+                    rabbitMQProducer.sendMessage(orderDTO);
```

Comment:
You send the order several times according to number of orderitem list, please do it outside the loop